game/scripts/vscripts/Sections/Chapter1/SectionLeveling.ts

```diff
@@ -43,18 +46,46 @@ const start = (complete: () => void) => {
 
         // Spawn purge and freeze player during initial dialog
         tg.immediate(_ => freezePlayerHero(true)),
-        tg.audioDialog(LocalizationKey.Script_1_Leveling_3, LocalizationKey.Script_1_Leveling_3, ctx => ctx[CustomNpcKeys.SlacksMudGolem]), // here comes pugna
-
-        tg.spawnUnit(CustomNpcKeys.PurgePugna, pugnaLocation, DotaTeam.BADGUYS, CustomNpcKeys.PurgePugna, true),
-        tg.immediate(ctx => getOrError(ctx[CustomNpcKeys.PurgePugna] as CDOTA_BaseNPC | undefined).SetAttackCapability(UnitAttackCapability.NO_ATTACK)),
-
+        tg.fork([
+            tg.seq([
+                tg.audioDialog(LocalizationKey.Script_1_Leveling_3, LocalizationKey.Script_1_Leveling_3, ctx => ctx[CustomNpcKeys.SlacksMudGolem]), // here comes pugna
+            ]),
+            tg.seq([
+                tg.spawnUnit(CustomNpcKeys.PurgePugna, pugnaLocation, DotaTeam.BADGUYS, CustomNpcKeys.PurgePugna, true),
+                tg.setCameraTarget(context => context[CustomNpcKeys.PurgePugna]),
+                tg.wait(FrameTime()),
```

Comment:
Why is this needed?

Reply:
Had to add this because the unit spawns and is ordered to move in the same frame, which causes it to not yet be "initialized" enough to get commands. This caused it to ignore the move command.